### PR TITLE
support cluster.newInstance

### PR DESCRIPTION
**Pull request by ronenhamias**

### CODE REVIEW

File: cluster/src/main/java/io/servicefabric/cluster/Cluster.java [hunk 1]
```diff
@@ -9,14 +9,24 @@
...
 	private final IGossipProtocol gossipProtocol;
 	private final IClusterMembership clusterMembership;
 
-	Cluster(ITransport transport, IFailureDetector failureDetector, IGossipProtocol gossipProtocol,
+	public static ICluster newInstance(String memberId, int port, String seedMembers){
+		String uri = "tcp://"+memberId+"@localhost:"+port;
+		return ClusterBuilder.newInstance(ClusterEndpoint.from(uri), seedMembers).build();
+	}
+	
+	public static ICluster newInstance(String memberId, int port){
```
Comment (antonkharenko) [inline cluster/src/main/java/io/servicefabric/cluster/Cluster.java]: Use ClusterBuilder for such stuff - this is package private class

File: cluster/src/main/java/io/servicefabric/cluster/Cluster.java [hunk 2]
```diff
@@ -9,14 +9,24 @@
 /**
  * @author Anton Kharenko
  */
-class Cluster implements ICluster {
+public class Cluster implements ICluster {
```
Comment (antonkharenko) [inline cluster/src/main/java/io/servicefabric/cluster/Cluster.java]: Remove public here

File: cluster/src/main/java/io/servicefabric/cluster/ClusterBuilder.java [hunk 3]
```diff
@@ -34,6 +34,16 @@ private ClusterBuilder(ClusterEndpoint localClusterEndpoint, String wellknownMem
 		this.wellknownMembers = wellknownMembers;
 	}
 
+	public static ICluster newInstance(String memberId, int port, String seedMembers){
+		String uri = "tcp://"+memberId+"@localhost:"+port;
```
Comment (artem-v) [inline cluster/src/main/java/io/servicefabric/cluster/ClusterBuilder.java]: Use ```ClusterEndpoint``` and ```TransportEndpoint``` api to build those objects. This line ```String uri = "tcp://"+memberId+"@localhost:"+port;``` is bad idea.

File: cluster/src/main/java/io/servicefabric/cluster/ClusterBuilder.java [hunk 4]
```diff
@@ -34,6 +34,16 @@ private ClusterBuilder(ClusterEndpoint localClusterEndpoint, String wellknownMem
...
+		return ClusterBuilder.newInstance(ClusterEndpoint.from(uri), seedMembers).build();
+	}
+	
+	public static ICluster newInstance(String memberId, int port){
+		String uri = "tcp://"+memberId+"@localhost:"+port;
```
Comment (artem-v) [inline cluster/src/main/java/io/servicefabric/cluster/ClusterBuilder.java]: See prev comment.

File: cluster/src/main/java/io/servicefabric/cluster/ClusterBuilder.java [hunk 5]
```diff
@@ -34,6 +34,16 @@ private ClusterBuilder(ClusterEndpoint localClusterEndpoint, String wellknownMem
 		this.wellknownMembers = wellknownMembers;
 	}
 
+	public static ICluster newInstance(String memberId, int port, String seedMembers){
+		String uri = "tcp://"+memberId+"@localhost:"+port;
+		return ClusterBuilder.newInstance(ClusterEndpoint.from(uri), seedMembers).build();
+	}
+	
+	public static ICluster newInstance(String memberId, int port){
```
Comment (antonkharenko) [inline cluster/src/main/java/io/servicefabric/cluster/ClusterBuilder.java]: How are you supposing to interpret absence of seed members ?

Comment (ronenhamias) [inline cluster/src/main/java/io/servicefabric/cluster/ClusterBuilder.java]: currently what happens is cluster will not start
i think its wrong
cluster should try and restart

**Comment by antonkharenko** [comment]
See it here https://github.com/servicefabric/servicefabric/blob/api-optimization/cluster/src/main/java/io/servicefabric/cluster/ClusterBuilder.java#L39

**Pull request closed without merging**